src/main/scala/firrtl_interpreter/repl/ReplConfig.scala

```diff
@@ -0,0 +1,113 @@
+// See LICENSE for license details.
+
+package firrtl_interpreter.repl
+
+import firrtl.AnnotationSeq
+import firrtl.annotations.NoTargetAnnotation
+import firrtl.options.{HasScoptOptions, OptionsView, RegisteredLibrary}
+import scopt.OptionParser
+
+case class ReplExecutionOptions(
+  firrtlSourceName : String = "",
+  scriptName       : String = "",
+  firrtlSource     : String = "",
+  useVcdScript     : Boolean = false,
+  vcdScriptOverride: String = "",
+  runScriptAtStart : Boolean = false,
+  outputFormat     : String = "d"
+)
+
+object ReplViewer {
+  implicit object ReplOptionsView extends OptionsView[ReplExecutionOptions] {
+    def view(options: AnnotationSeq): Option[ReplExecutionOptions] = {
+      val executionOptions = options.foldLeft(ReplExecutionOptions()) { (previousOptions, annotation) =>
+        annotation match {
+          case ReplSourceNameAnnotation(name)                  => previousOptions.copy(firrtlSourceName = name)
+          case ReplFirrtlAnnotation(name)                      => previousOptions.copy(firrtlSource = name)
+          case ReplScriptNameAnnotation(name)                  => previousOptions.copy(scriptName = name)
+          case ReplUseVcdScriptAnnotation                      => previousOptions.copy(useVcdScript = true)
+          case ReplRunScriptOnStartupAnnotation                => previousOptions.copy(runScriptAtStart = true)
+          case _ => previousOptions
+        }
+
+      }
+      Some(executionOptions)
+    }
+  }
+}
+
+sealed trait ReplOption extends HasScoptOptions
+
+case class ReplSourceNameAnnotation(name: String = "") extends NoTargetAnnotation with ReplOption {
+  def addOptions(p: OptionParser[AnnotationSeq]): Unit = p.opt[String]("irepl-source-file")
+    .abbr("ir-sf")
+    .action( (name, c) => c :+ ReplSourceNameAnnotation(name) )
+    .unbounded()
+    .text("firrtl source for repl to operate on")
+}
+
+case class ReplFirrtlAnnotation(name: String = "") extends NoTargetAnnotation with ReplOption {
+  def addOptions(p: OptionParser[AnnotationSeq]): Unit = p.opt[String]("irepl-source")
+    .abbr("ir-s")
+    .action( (firrtl, c) => c :+ ReplFirrtlAnnotation(firrtl) )
+    .unbounded()
+    .text("firrtl source for repl to operate on")
+}
+
+case class ReplScriptNameAnnotation(name: String = "") extends NoTargetAnnotation with ReplOption {
+  def addOptions(p: OptionParser[AnnotationSeq]): Unit = p.opt[String]("irepl-script-name")
+    .abbr("ir-sf")
+    .action( (name, c) => c :+ ReplScriptNameAnnotation(name) )
+    .unbounded()
+    .text("script file to load")
+}
+
+case object ReplUseVcdScriptAnnotation extends NoTargetAnnotation with ReplOption {
+  def addOptions(p: OptionParser[AnnotationSeq]): Unit = p.opt[Unit]("irepl-use-vcd")
+    .abbr("ir-sf")
```

Comment:
This is duplicated three times in total. Note: totally up to you if you even want to have short options. It gets complicated to manage them all...

Reply:
just sloppy. I have dusted off my old `grep ... | cut -d: -f1 | sort | uniq -c | sort -n` command and think I have fixed them all now